gdc_client/query/versions.py

```diff
@@ -30,14 +30,23 @@ def get_latest_versions(url, uuids, verify=True):
         resp = requests.post(versions_url, json={"ids": chunk}, verify=verify)
 
         # Parse the results of the chunked query.
-        for result in resp.json():
-            file_id = result.get("id")
-            uuid = result.get("latest_id")
-            if uuid:
-                latest_versions[file_id] = uuid
-            else:
-                # Might happen for legacy files
-                latest_versions[file_id] = file_id
+        try:
+            for result in resp.json():
+                file_id = result.get("id")
+                uuid = result.get("latest_id")
+                if uuid:
+                    latest_versions[file_id] = uuid
+                else:
+                    # Might happen for legacy files
+                    latest_versions[file_id] = file_id
+        except json.JSONDecodeError as e:
+            # json() will fail if response body is not valid JSON
+            logger.error(
+                "The following response content resulted from request {0} for ids {1}: \n{2}".format(
+                    versions_url, chunk, resp.content
+                )
+            )
+            raise e
```

Comment:
You should just use raise here to re-raise the exception. That way the original stack trace is preserved.

Reply:
the original stack trace is preserved either way: both "raise e" and "raise" give me the same output when testing. I think its only when you raise a new Exception that you lose the original stack trace